kernel/kernel-api/src/main/java/io/delta/kernel/expressions/predicates/Predicate.java

```diff
@@ -0,0 +1,172 @@
+ *    <li>Since version: 3.0.0</li>
+ *   </ul>
+ *  </li>
+ *  <li>Name: <code>&gt;</code>
+ *   <ul>
+ *    <li>SQL semantic: <code>expr1 &gt; expr2</code></li>
+ *    <li>Since version: 3.0.0</li>
+ *   </ul>
+ *  </li>
+ *  <li>Name: <code>&gt;=</code>
+ *   <ul>
+ *    <li>SQL semantic: <code>expr1 &gt;= expr2</code></li>
+ *    <li>Since version: 3.0.0</li>
+ *   </ul>
+ *  </li>
+ *  <li>Name: <code>ALWAYS_TRUE</code>
+ *   <ul>
+ *    <li>SQL semantic: <code>Constant expression whose value is `true`</code></li>
+ *    <li>Since version: 3.0.0</li>
+ *   </ul>
+ *  </li>
+ *  <li>Name: <code>ALWAYS_FALSE</code>
+ *   <ul>
+ *    <li>SQL semantic: <code>Constant expression whose value is `false`</code></li>
+ *    <li>Since version: 3.0.0</li>
+ *   </ul>
+ *  </li>
+ *  <li>Name: <code>AND</code>
+ *   <ul>
+ *    <li>SQL semantic: <code>expr1 AND expr2</code></li>
+ *    <li>Since version: 3.0.0</li>
+ *   </ul>
+ *  </li>
+ *  <li>Name: <code>OR</code>
+ *   <ul>
+ *    <li>SQL semantic: <code>expr1 OR expr2</code></li>
+ *    <li>Since version: 3.0.0</li>
+ *   </ul>
+ *  </li>
+ * </ol>
+ */
+public class Predicate extends ScalarExpression {
+    public Predicate(String name, List<Expression> children) {
+        super(name, children);
+        validatePredicateExpression(name, children);
+    }
+
+    @Override
+    public String toString() {
+        if (COMPARATOR_OPERATORS.contains(name)) {
+            return String.format("(%s %s %s)", children.get(0), name, children.get(1));
+        }
+        String args = children.stream().map(Object::toString)
+            .collect(Collectors.joining(","));
+        return String.format("%s(%s)", name, args);
+    }
+
+    private static final Set<String> COMPARATOR_OPERATORS =
+        Stream.of("=", ">", ">=", "<", "<=").collect(Collectors.toSet());
+    /**
+     * List of supported predicate expressions
+     */
+    private static final Set<String> PREDICATE_TYPES = Stream.of(
+        "=",
+        "<",
+        "<=",
+        ">",
+        ">=",
+        "AND",
+        "OR",
+        "ALWAYS_TRUE",
+        "ALWAYS_FALSE"
+    ).collect(Collectors.toSet());
+
+    /**
+     * Validate the predicate name is in the allowed list and expected children types.
+     *
+     * @param name Predicate expression name
+     * @param children List of input expressions to the predicate expression.
+     */
+    private static void validatePredicateExpression(String name, List<Expression> children) {
+        checkArgument(PREDICATE_TYPES.contains(name), "Unknown predicate expression: %s", name);
+        switch (name) {
+            case "AND": // fall through
+            case "OR":
+                validateBinaryArgs(name, children);
+                validatePredicateArg(name, children, 0);
+                validatePredicateArg(name, children, 1);
+                break;
+            case "=":
+            case ">":
+            case ">=":
+            case "<":
+            case "<=":
```

Comment:
nit: this is so painful in java 8 that we can collapse this into a expression / function call like Scala `case isComparator(_)` :(